### DISCUSSION TITLE
[4.0] Feature request: Limit folder size in media manager

### DISCUSSION THREAD

**Discussion by ot2sen** [joomla/joomla-cms#23961] [2. Cold Features]
For the current J3 and former series, the users haven´t had a "better practice" guide, to help them act in their own best interest when populating their media folder(s).

This feature request is based upon once again seeing a site with a gigantic amount of media files in the root of the /images folder. This last case of volunteer work showed approximately 1400 images in the root of /images, which means it takes ages to load and eat up system resources. Yes, they should have used fewer images and made use of strategic use of subfolders, but they didn´t and didn´t know it would be smarter.

Suggestion: Set a default folder size of the /images folder and each of its subfolders, to help users do a better practice. The default could be set at installation based on the optimal settings and system resources available at the users web hosting. 
At each upload of a new image, a check would then be in place to verify that ex. countSize of folder was within limits, or else it would suggest user create a new subfolder in order to keep the fastest and best user experience when adding images to their site.

Without such a limit people tend to use a practice that is not in their own best interest. We could help them, and thereby us in ensuring the image handling always stays fast and easy to deal with.

**Comment by infograf768** [comment]
Looked on the web. No idea how to define a folder limit size via php.

**Comment by dneukirchen** [comment]
we should not limit the folder size... users should be able to upload as many media files as they want... 

instead we should paginate the requests so that the load time is acceptable. 
on server side we need to implement paginated responses.
on client we need to implement infinite scroll or show a pager.

**Comment by brianteeman** [comment]
That's a much better solution and also one that doesn't break backwards compatibility on a site that already has "too many" images in the one folder

**Comment by ot2sen** [comment]
Pagination sounds good too, agree. Would also bring life back to those having used less good practice for years.

**Comment by ghost** [comment]
That's a much better solution #23261 combined with @dneukirchen proposal its the best

> 
> we should not limit the folder size... users should be able to upload as many media files as they want...
> 
> instead we should paginate the requests so that the load time is acceptable.
> on server side we need to implement paginated responses.
> on client we need to implement infinite scroll or show a pager.

**Comment by ghost** [comment]
@ot2sen can this Issue be closed?

**Comment by brianteeman** [comment]
no - this is a valid feature request

**Comment by ghost** [comment]
@brianteeman thanks for Info.
